src/common/makeOutcomeFailDependingOnOutput.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import {ISpawnResult} from "./node/childProcess";
+
+export type PatternToFailure = { [pattern: string]: string };
+
+/* This class will transform a spawn process to only succed if all defined success patterns
+   are found on stdout, and none of the failure patterns were found on stderr */
+export class MakeOutcomeFailDependingOnOutput {
+    private generatePatternsForSuccess: () => Q.Promise<string[]>;
+    private generatePatternToFailure: () => Q.Promise<PatternToFailure>;
+
+    private output = "";
+    private errors = "";
+
+    constructor(generatePatternsForSuccess: () => Q.Promise<string[]>, generatePatternToFailure: () => Q.Promise<PatternToFailure>) {
```

Comment:
In the case of iOS, we can only find out the name of the bundle only after the app has been build or something similar, which happens when we execute run-ios, so we don't know the patterns when we call this command, only after it's "finished".